presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataReader.java

```diff
@@ -248,22 +248,33 @@ private static KeyProvider toKeyProvider(DwrfProto.Encryption.KeyProvider keyPro
         }
     }
 
-    private static List<StripeInformation> toStripeInformation(List<DwrfProto.StripeInformation> types)
-    {
-        return ImmutableList.copyOf(Iterables.transform(types, DwrfMetadataReader::toStripeInformation));
+    private static List<StripeInformation> toStripeInformation(List<DwrfProto.StripeInformation> stripeInformations)
+    {
+        ImmutableList.Builder<StripeInformation> stripeInfoBuilder = ImmutableList.builder();
+        List<byte[]> previousKeysMetadata = ImmutableList.of();
+        for (DwrfProto.StripeInformation dwrfStripeInfo : stripeInformations) {
+            StripeInformation prestoStripeInfo = toStripeInformation(dwrfStripeInfo, previousKeysMetadata);
+            stripeInfoBuilder.add(prestoStripeInfo);
+            previousKeysMetadata = prestoStripeInfo.getKeyMetadata();
+        }
+        return stripeInfoBuilder.build();
     }
 
-    private static StripeInformation toStripeInformation(DwrfProto.StripeInformation stripeInformation)
+    private static StripeInformation toStripeInformation(DwrfProto.StripeInformation stripeInformation, List<byte[]> previousKeyMetadata)
     {
+        List<byte[]> keyMetadata = stripeInformation.getKeyMetadataList().stream()
+                .map(ByteString::toByteArray)
+                .collect(toImmutableList());
+        if (keyMetadata.size() == 0) {
+            keyMetadata = previousKeyMetadata;
```

Comment:
For the first stripe, ```previousKeyMetadata``` will be empty. Could we hit some edge case there?

Reply:
For that case, if keyMetadata is empty, then keyMetadata and previousKeyMetadata are the same (both an empty list), so it doesn't really matter which one we use.